doc/application/application.rst

```diff
@@ -369,6 +369,75 @@ again.
 
 .. _application_debugging:
 
+
+Custom Board Definition
+***********************
+
+In cases where the board or platform you are developing for is not yet supported
+by Zephyr you can add the board definition to your application and build for
+this board without having to add it to the Zephyr tree.
+
+The structure needed to support an out-of-tree board is similar to how boards
+are maintained in Zephyr making it easy to upstream a board to Zephyr once the
```

Comment:
Break into two sentences:

    The structure needed to support out-of-tree board development
    is similar to how boards are maintained in the Zephyr tree.  By using
    this structure, it will be much easier to upstream your board work into
    the Zephyr tree after your initial development is done.